src/test/java/org/broadinstitute/hellbender/utils/test/FuncotatorReferenceTestUtils.java

```diff
@@ -0,0 +1,108 @@
+
+import htsjdk.samtools.util.Lazy;
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.compress.utils.IOUtils;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.funcotator.FuncotatorTestConstants;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.function.Supplier;
+
+public class FuncotatorReferenceTestUtils {
+
+    private FuncotatorReferenceTestUtils() {}
+    private static Lazy<String> hg19Chr3Ref;
+    private static Lazy<String> hg19Chr19Ref;
+    private static Lazy<String> b37Chr3Ref;
+    private static Lazy<String> hg38Chr3Ref;
+
+    static {
+        hg19Chr3Ref = new Lazy<>(createInitializer(new File(FuncotatorTestConstants.HG19_CHR3_REFERENCE_FILE_NAME)));
+        hg19Chr19Ref = new Lazy<>(createInitializer(new File(FuncotatorTestConstants.HG19_CHR19_REFERENCE_FILE_NAME)));
+        b37Chr3Ref = new Lazy<>(createInitializer(new File(FuncotatorTestConstants.HG19_3_REFERENCE_FILE_NAME)));
+        hg38Chr3Ref = new Lazy<>(createInitializer(new File(FuncotatorTestConstants.HG38_3_REFERENCE_FILE_NAME)));
+    }
+
+    private static Supplier<String> createInitializer(final File fastaTarGz) {
+        return () -> FuncotatorReferenceTestUtils.extractFastaTarGzToTemp(fastaTarGz,
+                org.broadinstitute.hellbender.utils.io.IOUtils.createTempDir("funcotatorTmpFolder").toPath());
+    }
+    /**
+     * Extract a fasta tar.gz (should include only one fasta, the index, and a dict).  Though this is not enforced.
+     *
+     * Much of this code taken from: http://commons.apache.org/proper/commons-compress/examples.html
+     *
+     * @param fastaTarGz Should include one fasta file, the associated dict file and the associated index, though this is not enforced.  Never {@code null}
+     * @param destDir Where to store te untarred files.  Never {@code null}
+     * @return a path (as string) to the fasta file.  If multiple fasta files were in the tar gz, it will return that last one seen.
+     */
+    private static String extractFastaTarGzToTemp(final File fastaTarGz, final Path destDir) {
+        Utils.nonNull(fastaTarGz);
+        Utils.nonNull(destDir);
+        String result = null;
+        try (final ArchiveInputStream i = new TarArchiveInputStream(new GzipCompressorInputStream(new FileInputStream(fastaTarGz)))) {
+            ArchiveEntry entry = null;
+            while ((entry = i.getNextEntry()) != null) {
+                if (!i.canReadEntryData(entry)) {
+                    // log something?
```

Comment:
Yeah - we should log something in this case.  The tests could be affected by this, but may not be.

Reply:
Done